src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -480,8 +482,13 @@ public void buffer(SinkRecord sinkRecord) {
     buffer.add(sinkRecord);
   }
 
-  public long offset() {
-    return offset;
+  /**
+   * Get the offset that was last committed to HDFS, or null if nothing was committed yet.
+   *
+   * @return the last committed offset; may be null
+   */
+  public Long committedOffset() {
+    return committedOffset;
```

Comment:
The `offset` value was *almost* what we needed, but unfortunately it is modified with the first record the writer processes. This means that it does not always represent the last offset that was committed.

Rather than change the logic that relied upon this behavior, I chose to introduce a new `committedOffset` member to represent the last offset that was committed, and set it equal to `offset` whenever the writer commits its file.

Reply:
Interesting. I tried changing it locally, and it felt awkward (e.g., in the JavaDoc for `offsetToCommit()` because, for the most part, this `TopicPartitionWriter` uses "commit" specifically for committing to HDFS. `offsetToCommit()` just felt like it was the offset that had yet to be written to HDFS.

So I guess I prefer leaving it `committedOffset`.

Reply:
Not insisting here. Just wanted to make an observation. `Commit` is a very overloaded term in these connectors indeed.